lib/tasks/llm/llms/elevenlabs_s2s.js

```diff
@@ -244,13 +244,36 @@ class TaskLlmElevenlabs_S2S extends Task {
     /* tool calls */
     else if (type === 'client_tool_call') {
       this.logger.debug({evt}, 'TaskLlmElevenlabs_S2S:_onServerEvent - function_call');
-      if (!this.toolHook) {
+      const {tool_name: name, tool_call_id: call_id, parameters: args} = evt.client_tool_call;
+
+      const tools = await this.parent.getAvailableMcpTools();
```

Comment:
It seems we are waiting for the actual tool call to be made to establish the mcp client connection to the server and find out about the available service - this will add latency into the tool call.  I'm thinking we that if mcp servers are supplied we should resolve the services it offers right when the task starts?

Reply:
Fixed.